src/deprecated.jl

```diff
@@ -357,3 +357,87 @@ function by(d::AbstractDataFrame, cols::Any; sort::Bool=false, skipmissing::Bool
 end
 
 @deprecate groupvars(gd::GroupedDataFrame) groupcols(gd)
+
+export aggregate
+
+function aggregate(d::AbstractDataFrame, f::Any; sort::Bool=false)
+    df = select(d, names(d) .=> [f])
+    if sort
+        Base.depwarn("`aggregate` is deprecated. " *
+                     "Instead use `sort!(select(d, names(d) .=> f))`.", :aggregate)
+        sort!(df)
+    else
+        Base.depwarn("`aggregate` is deprecated. " *
+                     "Instead use `select(d, names(d) .=> f)`.", :aggregate)
+    end
+    return df
+end
+
+function aggregate(d::AbstractDataFrame, fs::AbstractVector; sort::Bool=false)
+    df = hcat([select(d, names(d) .=> [f]) for f in fs]..., makeunique=true)
+    if sort
+        Base.depwarn("`aggregate` is deprecated. Instead" *
+                     " use `sort!(select(d, [names(d) .=> f for f in fs]...))` " *
+                     "if functions in `fs` have unique names.", :aggregate)
+        sort!(df)
+    else
+        Base.depwarn("`aggregate` is deprecated. Instead use " *
+                     "`select(d, [names(d) .=> f for f in fs]...)` if functions " *
+                     "in `fs` have unique names.", :aggregate)
+    end
+    return df
+end
+
+function aggregate(gd::GroupedDataFrame, f::Any; sort::Bool=false)
+    df = combine(gd, valuecols(gd) .=> [f])
+    if sort
+        Base.depwarn("`aggregate` is deprecated. Instead use" *
+                     " `df = combine(gd, valuecols(gd) .=> f); " *
+                     "sort!(df, setdiff(names(df), groupcols(gd)))`.",
+                     :aggregate)
+        sort!(df, setdiff(names(df), groupcols(gd)))
+    else
+        Base.depwarn("`aggregate` is deprecated. Instead use" *
+                     " `combine(gd, [valuecols(gd) .=> f for f in fs]...)`",
```

Comment:
Maybe this would be more explicit? People might not understand what `fs` refers to. Same elsewhere.
```suggestion
        Base.depwarn("`aggregate(gd, fs, sort=$sort)` is deprecated. Instead use" *
                     " `combine(gd, [valuecols(gd) .=> f for f in fs]...)`",
```